lib/sycamore/sycamore/connectors/common.py

```diff
@@ -88,49 +112,86 @@ def flatten_data(
     return items
 
 
-def unflatten_data(data: dict[str, Any], separator: str = ".") -> dict[Any, Any]:
-    result: dict[Any, Any] = {}
+def unflatten_data(data: dict[Any, Any], separator: str = ".") -> dict[Any, Any]:
+    """
+    Unflattens a dictionary with keys that contain separators into a nested dictionary. The separator can be escaped,
+    and if there are integer keys in the path, the result will be a list instead of a dictionary.
+    """
 
-    def parse_key(key: str) -> list:
-        # Handle escaped separator
+    def split_key(key: str, separator: str = ".") -> list[str]:
```

Comment:
why not just use str.split(sep=...)?

Reply:
This allows us to ignore escaped characters. It's behavior that was present in the previous function and we have tests that check for this so didn't want to change it.